src/Tribe/Linked_Posts.php

```diff
@@ -0,0 +1,812 @@
+				$posts[ $post_type ] = $this->get_linked_posts_by_post_type( $post_id, $post_type );
+			}
+		} else {
+			$post_type = Tribe__Events__Main::POSTTYPE;
+			$posts[ $post_type ] = $this->get_linked_posts_by_post_type( $post_id, $post_type );
+		}
+
+		/**
+		 * Filters the collection of linked posts for the provided post id
+		 *
+		 * @since 4.2
+		 *
+		 * @param array Collection of posts linked to the post id
+		 * @param int Post ID of the post being looked at
+		 */
+		return apply_filters( 'tribe_events_get_linked_posts', $posts, $post_id );
+	}
+
+	/**
+	 * Returns whether or not there are linked posts of the specified post type
+	 *
+	 * @since 4.2
+	 *
+	 * @param int $post_id Post ID of the object
+	 * @param string $post_type Post type of linked posts to look for
+	 *
+	 * @return boolean
+	 */
+	public function has_linked_posts_by_post_type( $post_id, $post_type ) {
+		$has_linked_posts = ! empty( $this->get_linked_posts_by_post_type( $post_id, $post_type ) );
+
+		/**
+		 * Filters whether or not a post has any linked posts of a given type
+		 *
+		 * @since 4.2
+		 *
+		 * @param boolean Whether or not there are any linked posts for the given post by the given post type
+		 * @param int Post ID being looked at
+		 * @param string Post type of linked posts
+		 */
+		$has_linked_posts = apply_filters( 'tribe_events_has_linked_posts_by_post_type', $has_linked_posts, $post_id, $post_type );
+
+		return $has_linked_posts;
+	}
+
+	/**
+	 * Returns linked posts of the specified post type
+	 *
+	 * @since 4.2
+	 *
+	 * @param int $post_id Post ID of the object
+	 * @param string $post_type Post type of linked posts to look for
+	 *
+	 * @return array
+	 */
+	public function get_linked_posts_by_post_type( $post_id, $post_type ) {
+		$result = array();
+
+		if ( $linked_post_ids = get_post_meta( $post_id, $this->get_meta_key( $post_type ) ) ) {
+			$result = $this->get_linked_post_info( $post_type, array(), $linked_post_ids );
+		}
+
+		/**
+		 * Filters the linked posts of a given type for the given post
+		 *
+		 * @since 4.2
+		 *
+		 * @param array Linked posts for the given post by the given post type
+		 * @param int Post ID being looked at
+		 * @param string Post type of linked posts
+		 */
+		return apply_filters( 'tribe_events_get_linked_posts_by_post_type', $result, $post_id, $post_type );
+	}
+
+	/**
+	 * Returns whether or not there are any linked post types
+	 *
+	 * @since 4.2
+	 *
+	 * @return boolean
+	 */
+	public function has_linked_post_types() {
+		return ! empty( $this->linked_post_types );
+	}
+
+	/**
+	 * Returns the linked post types
+	 *
+	 * @since 4.2
+	 *
+	 * @return array
+	 */
+	public function get_linked_post_types() {
+		return (array) $this->linked_post_types;
+	}
+
+	/**
+	 * Get Linked Post info
+	 *
+	 * @param string $linked_post_type Post type of linked post
+	 * @param array $args
+	 * @param int $linked_post_id post id
+	 *
+	 * @return WP_Query->posts || array()
```

Comment:
I actually quite like this but might it be better to follow the normal pattern with an actual type like `@return array` optionally followed by a description such as `list of WP_Post objects (may be empty)`? Just thinking the current approach used here may confuse intellisense-type features for various IDE users.